Shopify/dawn
# [Product Cards] Add image masks

### PR Summary: 

Adds seven image masks to product cards (Arch, Blobs, Chevron Left, Chevron Right, Diamond Parallelogram, Diamond, Round). 

Examples: 

https://user-images.githubusercontent.com/1202812/218860178-9bc8691e-cb7d-4dee-b77f-9ab7efb1dbe8.mp4


### Why are these changes introduced?

These shapes introduce more design possibilities to our merchants. 

### What approach did you take?

All of these are implemented using CSS `clip-path()`. The Arch style is created using an SVG mask, while the others are powered by simple `polygon()` and `circle()` CSS functions. Many shapes include simple, pure-css hover animations. 

The most complicated of these implementations is the Blobs shape. There are six total blob shapes, and the PR uses `nth-child()` to approximate a random-ish pattern when they're arranged in a product grid. The blobs have been implemented using `polygon()` instead of using simpler SVG paths, because SVG paths cannot be animated natively with CSS. This approach works well — the main downfall is that `polygon()` supports points but not curves, and blobs are made of curves. To avoid the shapes looking pointy at large sizes, each blob has 120 points. This gets a little verbose. I've assigned each blob shape to a CSS variable to compensate and maintain code readability.

### Other considerations

- I'd love another developer perspective on the blobs implementation. If we want them to animate (which I think is the most fun part, personally), this seemed like the leanest approach to me.
- As noted above, many of these shapes include some light hover effects. I did not include checkboxes to turn these off, but we can do that if folks feel strongly about it.
- The hover effects all respect the visitor's `prefers-reduced-motion` preferences. 
- The border and shadow product card controls aren’t effective in these new shapes, because the existing implementations don’t work for clip-paths. We could [build in alternatives](https://css-tricks.com/using-box-shadows-and-clip-path-together/), but that feels overcomplicated (and also not necessarily critical for a v1). For now, I’ve added a note next to the Style control that introduces this caveat. (FWIW, the border radius does partially function in some circumstances, and is [kinda cool](https://screenshot.click/14-42-floi4-5gxfu.mp4).). 

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 | Implement product card shape settings on a section level. | Implement one global setting that turns the shapes on for all product cards on the store. | The shapes are relatively opinionated, so we expect merchants will want to use them sparingly rather than turn them on everywhere all at once. | If a merchant does want to use the shapes globally, they'll need to track down all instances of product cards and change the shapes for those individual Sections. |

### Visual impact on existing themes

These shapes are off by default, and are optional. I don't expect any issues using them on other Dawn-based themes. 

### Testing steps/scenarios

1. [Visit the demo store.](https://os2-demo.myshopify.com/admin/themes/139389894678/editor)
2. Add a **Featured Collection** section. 
3. In the Product Cards section of the settings panel, use the "Image Shape" dropdown to change the image shape.
4. Try all 7 of the supplied image shapes. 
5. Change other Section settings, to ensure they don't interfere with the image shape. 
6. Scale the viewport from desktop down to mobile, ensuring that the shapes behave as intended with no clipping. 
7. Repeat steps 2-6 with the following additional sections: **Product Grid**, **Search Results**, and **Related Products**.
8. Visit Theme Settings, and adjust settings for the product cards (Style, Padding, Color Scheme, etc.) and ensure that the cards don't appear broken. 

### Checklist
